### ISSUE TITLE
NSGenericException Collection was mutated while being enumerated

### ISSUE THREAD

**Issue by cristianoccazinsp** [bugsnag/bugsnag-js#1042]
### Describe the bug
Reported recently, looks like an internal bugsnag error, I may be wrong. See below for stack traces and info.

### Environment
* Bugsnag version: "@bugsnag/react-native": "7.3.3"
* CocoaPods version (if any): (pod --version) 1.9.1
* Xcode version (if any): 11.7
* iOS/Android version(s): iOS 13.6.x
* simulator/emulator or physical device: device
* Using Hermes on Android?: No
* debug mode or production?: Production

### Example code snippet 

The following was reported by bugsnag:
```
NSGenericException *** Collection <__NSArrayM: 0x282cce190> was mutated while being enumerated. 
    Frameworks/CoreFoundation.framework/CoreFoundation ___exceptionPreprocess
    /usr/lib/libobjc.A.dylib _objc_exception_throw
    Frameworks/CoreFoundation.framework/CoreFoundation ___NSFastEnumerationMutationHandler
    Tenant Move-In _mh_execute_header
    Tenant Move-In _mh_execute_header
    Tenant Move-In _mh_execute_header
    Tenant Move-In _mh_execute_header
    Tenant Move-In _mh_execute_header
    Frameworks/CoreFoundation.framework/CoreFoundation ___invoking___
    Frameworks/CoreFoundation.framework/CoreFoundation -[NSInvocation invoke]
    Frameworks/CoreFoundation.framework/CoreFoundation -[NSInvocation invokeWithTarget:]
    Tenant Move-In facebook::react::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
    Tenant Move-In facebook::react::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
    Tenant Move-In facebook::react::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
    /usr/lib/system/libdispatch.dylib __dispatch_call_block_and_release
    /usr/lib/system/libdispatch.dylib __dispatch_client_callout
    /usr/lib/system/libdispatch.dylib __dispatch_lane_serial_drain
    /usr/lib/system/libdispatch.dylib __dispatch_lane_invoke
    /usr/lib/system/libdispatch.dylib __dispatch_workloop_worker_thread
    /usr/lib/system/libsystem_pthread.dylib __pthread_wqthread
    /usr/lib/system/libsystem_pthread.dylib _start_wqthread
```

Following this, the XCode reported stack trace ends up in bugsnag code, as follows:

<img width="457" alt="Screen Shot 2020-09-07 at 22 23 00" src="https://user-images.githubusercontent.com/48869228/92423814-c30c4a00-f158-11ea-83f2-2a4912d6b41b.png">

<img width="774" alt="Screen Shot 2020-09-07 at 22 23 15" src="https://user-images.githubusercontent.com/48869228/92423821-cef80c00-f158-11ea-9cee-de6fb4f2449e.png">

Looking at the code that XCode reported the crash, it really makes sense since it comes from an array iteration.

**Comment by johnkiely1** [comment]
This has been transferred to the bugsnag-js repo as the Bugsnag react native library is now part of this repository (as of v7.3).

**Comment by cristianoccazinsp** [comment]
Have you guys confirmed the crash is in fact from bugsnag?  It is starting to ramp up in the logs and is becoming more frequent.

**Comment by phillipsam** [comment]
Hi @cristianoccazinsp - we are hoping to release a fix for this issue in the next day or so

**Comment by steve-nester-uk** [comment]
Fixed in [v7.3.5](https://github.com/bugsnag/bugsnag-js/releases/tag/v7.3.5)
